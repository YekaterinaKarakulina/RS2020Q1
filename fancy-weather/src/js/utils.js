function createDomElement(elementName, className) {
  const element = document.createElement(elementName);
  element.className = className;
  return element;
}

function translateCoordinates(lat, lng) {
  let degrees = 0;
  let minutes = 0;
  let seconds = 0;
  let latSign = 'N';
  let lngSign = 'E';
  let latitude = lat;
  let longitude = lng;
  if (lat < 0) {
    latSign = 'S';
    latitude = Math.abs(lat);
  }
  if (lng < 0) {
    lngSign = 'W';
    longitude = Math.abs(lng);
  }

  degrees = Math.floor(latitude);
  let remainder = latitude - degrees;
  minutes = Math.floor(remainder * 60);
  remainder = remainder * 60 - minutes;
  seconds = Math.floor(remainder * 60);
  latitude = `${degrees}°${minutes}'${seconds}''${latSign}`;

  degrees = Math.floor(longitude);
  remainder = longitude - degrees;
  minutes = Math.floor(remainder * 60);
  remainder = remainder * 60 - minutes;
  seconds = Math.floor(remainder * 60);
  longitude = `${degrees}°${minutes}'${seconds}''${lngSign}`;

  return { latitude, longitude };
}


/* function translateCToF(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent.slice(0, -1);
    const translatedTemp = (9 / 5) * currentT + 32;
    tempElem.textContent = `${Math.round(translatedTemp)}°`;
  });
  const realFeelTempElem = document.querySelector('.realFeelTemperature');
  const currentRealFeelTemp = realFeelTempElem.textContent.split(' ')[1].slice(0, -1);
  const translatedRealFeelTemp = (9 / 5) * currentRealFeelTemp + 32;
  realFeelTempElem.textContent = `REALFEEL ${Math.round(translatedRealFeelTemp)}°`;
}

function translateFToC(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent.slice(0, -1);
    const translatedTemp = (5 / 9) * (currentT - 32);
    tempElem.textContent = `${Math.round(translatedTemp)}°`;
  });
  const realFeelTempElem = document.querySelector('.realFeelTemperature');
  const currentRealFeelTemp = realFeelTempElem.textContent.split(' ')[1].slice(0, -1);
  const translatedRealFeelTemp = (5 / 9) * (currentRealFeelTemp - 32);
  realFeelTempElem.textContent = `REALFEEL ${Math.round(translatedRealFeelTemp)}°`;
} */


function transferCelsiusToFahrenheit(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent;
    const translatedTemp = (9 / 5) * currentT + 32;
    tempElem.textContent = Math.round(translatedTemp);
  });
}

function transferFahrenheitToCelsius(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent;
    const translatedTemp = (5 / 9) * (currentT - 32);
    tempElem.textContent = Math.round(translatedTemp);
  });
}

function transferTemperature() {
  const tempElements = document.querySelectorAll('.temperature');

  console.log(localStorage.getItem('isCelsius'));
  if (localStorage.getItem('isCelsius') === 'true') {
    console.log('yes');
    localStorage.setItem('isCelsius', 'false');
    transferCelsiusToFahrenheit(tempElements);
  } else {
    console.log('no');
    localStorage.setItem('isCelsius', 'true');
    transferFahrenheitToCelsius(tempElements);
  }
}

export { createDomElement, translateCoordinates, transferTemperature };
