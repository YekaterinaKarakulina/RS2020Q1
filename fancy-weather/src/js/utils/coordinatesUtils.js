
function translateCoordinate(coordinate) {
  let geoCoordinate = coordinate;
  if (coordinate < 0) {
    geoCoordinate = Math.abs(coordinate);
  }
  const degrees = Math.floor(geoCoordinate);
  let remainder = geoCoordinate - degrees;
  const minutes = Math.floor(remainder * 60);
  remainder = remainder * 60 - minutes;
  const seconds = Math.floor(remainder * 60);
  return `${degrees}Â°${minutes}'${seconds}''`;
}

export default function translateCoordinates(lat, lng) {
  let latSign = 'N';
  let lngSign = 'E';

  if (lat < 0) {
    latSign = 'S';
  }
  if (lng < 0) {
    lngSign = 'W';
  }

  const latitude = `${translateCoordinate(lat)}${latSign}`;
  const longitude = `${translateCoordinate(lng)}${lngSign}`;

  return { latitude, longitude };
}
