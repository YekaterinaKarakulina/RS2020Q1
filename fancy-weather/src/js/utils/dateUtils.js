import { weekDaysFull } from '../data/data';

function getCurrentDate(appObject) {
  const localeStrDate = new Date().toLocaleString('en-US', { timeZone: `${appObject.timezone}` });
  const currentDate = new Date(Date.parse(localeStrDate));
  return currentDate;
}

function getWeekDay(index) {
  if (index < weekDaysFull.length) {
    return weekDaysFull[index];
  }
  return weekDaysFull[index % 7];
}

export { getCurrentDate, getWeekDay };
