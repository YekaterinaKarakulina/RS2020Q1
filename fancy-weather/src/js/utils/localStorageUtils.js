export default function getLanguageAbbreviation(language) {
  let translateLang = '';
  switch (language) {
    case 'Russian':
      translateLang = 'ru';
      break;
    case 'Belorussian':
      translateLang = 'be';
      break;
    default:
      translateLang = 'en';
      break;
  }
  return translateLang;
}
