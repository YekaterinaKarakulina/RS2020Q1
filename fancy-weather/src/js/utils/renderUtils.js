function createDomElement(elementName, className) {
  const element = document.createElement(elementName);
  element.className = className;
  return element;
}

function createDomElementWithDataAttr(elementName, className, dataAttr, dataAttrValue) {
  const element = document.createElement(elementName);
  element.className = className;
  element.dataset[dataAttr] = dataAttrValue;
  return element;
}

export { createDomElement, createDomElementWithDataAttr };
