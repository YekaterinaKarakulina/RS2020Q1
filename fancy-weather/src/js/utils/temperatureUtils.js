function transferCelsiusToFahrenheit(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent;
    const translatedTemp = (9 / 5) * currentT + 32;
    tempElem.textContent = Math.round(translatedTemp);
  });
}

function transferFahrenheitToCelsius(tempElements) {
  tempElements.forEach((el) => {
    const tempElem = el;
    const currentT = el.textContent;
    const translatedTemp = (5 / 9) * (currentT - 32);
    tempElem.textContent = Math.round(translatedTemp);
  });
}

function transferTemperature(tempElements) {
  if (localStorage.getItem('temp') === 'isCelsius') {
    localStorage.setItem('temp', 'isFahrenheit');
    transferCelsiusToFahrenheit(tempElements);
  } else if (localStorage.getItem('temp') === 'isFahrenheit') {
    localStorage.setItem('temp', 'isCelsius');
    transferFahrenheitToCelsius(tempElements);
  }
}

export { transferTemperature, transferCelsiusToFahrenheit, transferFahrenheitToCelsius };
