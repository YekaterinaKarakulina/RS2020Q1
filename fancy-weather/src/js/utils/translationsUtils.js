async function getTranslations(url) {
  const res = await fetch(url);
  const data = await res.json();
  return data;
}

async function translateElement(language, dataAttr, elementToTranslate, elementAttr) {
  const urlToI18NFile = `/i18n/${language}.json`;
  const data = await getTranslations(urlToI18NFile);
  const element = elementToTranslate;
  element[elementAttr] = data[element.dataset[dataAttr]];
}

async function translateElements(language, dataAttr, elementsToTranslate, elementAttr) {
  elementsToTranslate.forEach((element) => {
    translateElement(language, dataAttr, element, elementAttr);
  });
}

async function switchLanguage(language) {
  const staticElements = document.querySelectorAll('[data-i18n]');
  translateElements(language, 'i18n', staticElements, 'textContent');

  const searchButtonElement = document.querySelector('[data-searchbtn]');
  translateElement(language, 'searchbtn', searchButtonElement, 'textContent');

  const placeholderElements = document.querySelectorAll('[data-placeholder]');
  translateElements(language, 'placeholder', placeholderElements, 'placeholder');
}

async function transferLanguageBeforeRendering(language, fragment) {
  const staticElements = fragment.querySelectorAll('[data-i18n]');
  translateElements(language, 'i18n', staticElements, 'textContent');

  const searchButtonElement = document.querySelector('[data-searchbtn]');
  translateElement(language, 'searchbtn', searchButtonElement, 'textContent');

  const placeholderElements = document.querySelectorAll('[data-placeholder]');
  translateElements(language, 'placeholder', placeholderElements, 'placeholder');
}

export { switchLanguage, transferLanguageBeforeRendering };
