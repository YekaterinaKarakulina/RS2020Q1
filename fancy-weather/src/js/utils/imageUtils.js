import { getCurrentDate } from './dateUtils';

export default function getKeywordsForImgAPI(appObject) {
  const currentDate = getCurrentDate(appObject);
  const month = currentDate.getMonth();
  const hours = currentDate.getHours();
  let season = '';
  let dayTime = '';
  if (month > 1 && month < 5) {
    season = (appObject.lat < 0) ? 'autumn' : 'spring';
  } else if (month > 4 && month < 8) {
    season = (appObject.lat < 0) ? 'winter' : 'summer';
  } else if (month > 7 && month < 11) {
    season = (appObject.lat < 0) ? 'spring' : 'autumn';
  } else {
    season = (appObject.lat < 0) ? 'summer' : 'winter';
  }
  if (hours > 5 && hours < 12) {
    dayTime = 'morning';
  } else if (hours >= 12 && hours < 18) {
    dayTime = 'day';
  } else if (hours >= 18 && hours < 24) {
    dayTime = 'evening';
  } else {
    dayTime = 'night';
  }
  const imgKeywords = `${season} ${dayTime}`;
  return imgKeywords;
}

// async function getAndRenderNewImg(imgProperties) {
//   const linkToImg = await getImage(imgProperties);
//   renderImage(linkToImg);
// }
