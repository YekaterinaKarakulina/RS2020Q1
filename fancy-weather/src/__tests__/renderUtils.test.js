import { createDomElement, createDomElementWithDataAttr } from '../js/utils/renderUtils';

describe('createDomElement', () => {
  it('Should return DOM element with className', () => {
    const result = createDomElement('div', 'className');
    expect(result).toMatchSnapshot();
  });
});

describe('createDomElement', () => {
  it('Should return DOM element with data attribute', () => {
    const result = createDomElementWithDataAttr('div', 'className', 'i18n', 'temp');
    expect(result).toMatchSnapshot();
  });
});
