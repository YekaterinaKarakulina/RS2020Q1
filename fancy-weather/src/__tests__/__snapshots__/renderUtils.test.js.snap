// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDomElement Should return DOM element with className 1`] = `
<div
  class="className"
/>
`;

exports[`createDomElement Should return DOM element with data attribute 1`] = `
<div
  class="className"
  data-i18n="temp"
/>
`;
